### PR TITLE
API-1448: [PoC] - Find a way to observe the event dispatcher in behat legacy

### PR DESCRIPTION
**Pull request by tomglvng**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/legacy/features/pim/enrichment/product/import/import_products.feature [hunk 1]
```diff
@@ -31,6 +31,7 @@ Feature: Import products coming from an external application
     And product "SKU-007" should be enabled
     And the english localizable value name of "SKU-001" should be "Donec"
     And the english tablet description of "SKU-002" should be "Pellentesque habitant morbi tristique senectus et netus et malesuada fames"
+    And 10 events of type "product.created" have been raised
```
Comment (tomglvng) [inline tests/legacy/features/pim/enrichment/product/import/import_products.feature:34]: should have been raised

**Review by willy-ahva — APPROVED** [review]

File: tests/legacy/features/Context/ConnectivityContext.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        $this->transport = $transport;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function setKernel(KernelInterface $kernel)
+    {
+        $this->kernel = $kernel;
+        self::$kernelRootDir = $kernel->getRootDir();
```
Comment (tseho) [inline tests/legacy/features/Context/ConnectivityContext.php:34]: to remove, it's not used

File: config/packages/behat/messenger.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ framework:
                 options:
                     project_id: '%env(GOOGLE_CLOUD_PROJECT)%'
                     topic_name: '%env(PUBSUB_TOPIC_BUSINESS_EVENT)%'
+                    subscription_name: '%env(PUBSUB_SUBSCRIPTION_BUSINESS_EVENT)%'
```
Comment (tseho) [inline config/packages/behat/messenger.yml:10]: Without a subscription defined, the GPSTransport does not have a receiver and does not allow to consume messages from the `business_event` queue.

File: tests/legacy/features/Context/ConnectivityContext.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        while (true) {
+            $envelopes = $this->transport->get();
+            if (empty($envelopes)) {
+                break;
+            }
+            foreach ($envelopes as $envelope) {
+                $this->transport->ack($envelope);
+            }
+            $this->envelopes = array_merge($this->envelopes, $envelopes);
```
Comment (tseho) [inline tests/legacy/features/Context/ConnectivityContext.php:50]:
```suggestion
            foreach ($envelopes as $envelope) {
                $this->transport->ack($envelope);
                $this->envelopes[] = $envelope;
            }
```
It's only a detail but it's more performant without array_merge.

**Review by tseho — APPROVED** [review]

**Pull request merged**